invenio_accounts/tokens.py

```diff
@@ -0,0 +1,212 @@
+            data = self.load_token(token)
+
+            # Compare expected data with data in token.
+            if expected_data:
+                for k in expected_data:
+                    if expected_data[k] != data["data"].get(k):
+                        return None
+            return data
+        except BadData:
+            return None
+
+    def load_token(self, token, force=False):
+        """Load data in a token.
+
+        :param token: Token to load.
+        :param force: Load token data even if signature expired.
+                      Default: False.
+        """
+        try:
+            data = self.loads(token)
+        except SignatureExpired as e:
+            if not force:
+                raise
+            data = e.payload
+
+        del data["rnd"]
+        return data
+
+
+class EncryptedTokenMixIn(TokenMixin):
+
+    """Mix-in class for token serializers that generate encrypted tokens."""
+
+    @property
+    def engine(self):
+        """Get cryptographic engine."""
+        if not hasattr(self, '_engine'):
+            from cryptography.fernet import Fernet
+            from cryptography.hazmat.backends import default_backend
+            from cryptography.hazmat.primitives import hashes
+
+            digest = hashes.Hash(hashes.SHA256(), backend=default_backend())
+            digest.update(cfg['SECRET_KEY'].encode('utf8'))
+            fernet_key = urlsafe_b64encode(digest.finalize())
+            self._engine = Fernet(fernet_key)
+        return self._engine
+
+    def create_token(self, obj_id, extra_data):
+        """Create a token referencing the object id with extra data."""
+        return self.engine.encrypt(
+            super(EncryptedTokenMixIn, self).create_token(obj_id, extra_data)
+        )
+
+    def load_token(self, token, force=False):
+        """Load data in a token.
+
+        :param token: Token to load.
+        :param force: Load token data even if signature expired.
+                      Default: False.
+        """
+        return super(EncryptedTokenMixIn, self).load_token(
+            self.engine.decrypt(token), force=force
+        )
+
+
+class EmailConfirmationSerializer(TimedJSONWebSignatureSerializer, TokenMixin):
+
+    """Serializer for email confirmation link tokens.
+
+    Depends upon the secrecy of ``SECRET_KEY``. Tokens expire after a specific
+    time (defaults to ``ACCOUNTS_CONFIRMLINK_EXPIRES_IN``). The
+    access request id as well as the email address is stored in the token
+    together with a random bit to ensure all tokens are unique.
+    """
+
+    def __init__(self, expires_in=None):
+        """Initialize underlying TimedJSONWebSignatureSerializer."""
+        dt = expires_in or cfg.get('ACCOUNTS_CONFIRMLINK_EXPIRES_IN')
+
+        super(EmailConfirmationSerializer, self).__init__(
+            cfg['SECRET_KEY'],
+            expires_in=dt,
+            salt='accounts-email',
+        )
+
+
+class SecretLinkSerializer(JSONWebSignatureSerializer,
```

Comment:
Are you using the SecretLinkSerializers at all? Otherwise I think we can just remove them.